pyccel/ast/cwrapper.py

```diff
@@ -495,9 +528,44 @@ def pybool_to_bool(cast_function_name):
     NativeReal() : PyFloat_AsDouble,
 }
 
-check_type_registry = {
+check_type_registry  = {
     NativeInteger(): 'PyLong_Check',
     NativeComplex() : 'PyComplex_Check',
     NativeReal() : 'PyFloat_Check',
     NativeBool() : 'PyBool_Check',
 }
+
+# Needed to check for numpy arguments type
+Numpy_Bool_ref = Variable(dtype=NativeVoid(),  name = 'Bool')
+Numpy_Byte_ref = Variable(dtype=NativeVoid(),  name = 'Byte')
+Numpy_Short_ref = Variable(dtype=NativeVoid(),  name = 'Short')
+Numpy_Int_ref = Variable(dtype=NativeVoid(),  name = 'Int')
+Numpy_Long_ref = Variable(dtype=NativeVoid(),  name = 'Long')
+Numpy_Float_ref = Variable(dtype=NativeVoid(),  name = 'Float')
+Numpy_Double_ref = Variable(dtype=NativeVoid(),  name = 'Double')
+Numpy_Complex_ref = Variable(dtype=NativeVoid(),  name = 'ComplexFloating')
```

Comment:
Are these sufficiently specific? Especially `ComplexFloating`?
I found a doc here:
https://numpy.org/devdocs/reference/arrays.scalars.html
which seems to indicate that `ComplexFloating` is abstract.

Reply:
It is not sufficient. We need `CFloat` and `CDouble`